src/main/java/com/faforever/client/replay/ReplayDetailController.java

```diff
@@ -321,6 +315,22 @@ private void populateTeamsContainer() {
     }));
   }
 
+  private void configureRatingControls() {
+    if (!replay.getValidity().equals(Validity.VALID)) {
+      showRatingChangeButton.setVisible(false);
+      notRatedReasonLabel.setVisible(true);
+      notRatedReasonLabel.setText(i18n.get("game.reasonNotValid", i18n.get(i18n.get("game.reasonNotValid.keyFormat", replay.getValidity().ordinal()))));
```

Comment:
Also you got to have a fallback. Maybe somebody adds a new validity reason to the server one day. I suggest if the i18n key does not exist just take the getValidity().name() and say it is the reason.

Reply:
On the other hand then parsing the json would not work so maybe that is not needed. But safety messure would not hurt...